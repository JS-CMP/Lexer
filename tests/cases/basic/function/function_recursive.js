function b(depth) {
    a(depth);
}

function a(depth) {
    if (depth == 0) {
        console.log('depth is 0');
        return 0;
    }
    return b(depth - 1);
}

a(3);

