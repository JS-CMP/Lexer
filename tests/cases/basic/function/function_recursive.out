#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"

JS::Any b(JS::Any);
JS::Any a(JS::Any);

JS::Any b(JS::Any depth) {a(depth);return JS::Any();}
JS::Any a(JS::Any depth) {if (depth==JS::Any(0)){console["log"](JS::Any("depth is 0"));return JS::Any(0);};return b(depth-JS::Any(1));}
int main() {a(JS::Any(3));}
